rfcs/1-declarative-configuration.md

```diff
@@ -0,0 +1,127 @@
+# Declarative Configuration
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Implementation Details](#implementation-details)
+  - [Configuration example](#examples)
+- [Test Plan](#testrelease-plan)
+<!-- /toc -->
+
+
+## Summary
+
+This proposal replaces environment variable based configuration being used in Make+Magefiles in favor of using declarative yaml based configuration.
+
+## Motiviation
+
+As we've been writing `devbase`, and other tooling, we've identified that trying to codify all options for tooling to environment variables, bespoke configuration files, and other obscure means has made it difficult for us to provide options and maintain a common set of toolign.
+
+This, combined with a lack of documentation into all of these options, has laregely made `devbase` unconsumable outside of bootstrap/stencil. Thinking of `devbase` as it's own framework to build software makes it more usable in non-stencil projects (like we do already today w/ submodules)
+### Goals
+
+ - Documentation for all options in configuration file(s)
+ - Standard location for all configuration for tooling provided by devbase, et. al
+ - Documentation mapping 1:1 with Magefile target, with framework to consume/declare it
+ - Documentation on how to write Magefile targets, if the framework docs aren't self-explanatory enough
+ - Support for usage outside of stencil-base, et. al
+ - Not breaking, while we could do a lot more if we did a breaking release, breaking _all_ of the existing releasing tooling isn't optimal
+
+### Non-goals
+
+ - Expose new functionality, this would be nice to have but would balloon the amount of work. We're targeting 1:1 compatibility with features/configuration already exposed today, with nice-to-haves being limited.
+ - Migrating `Makefile` to `Magefile`, while it'd be nice to migrate all of them over, that's also an amount of work to do. While moving to `Magefile` would be good, we shouldn't try to move everything to Go at the same time. (TL;DR: Wrap shell were needed)
+
+## Proposal
+
+The high level proposal here is generally covered in the [Summary](#summary) and [Moviation](#motiviation) sections.
+
+### Implementation Details
+
+#### Repository Configuration
+
+Configuration would live in the root of repository consuming `devbase`, in `.devbase`.
+
+Each configuration file would match the relevant `mage <target>` target, e.g.
+
+```bash
+$ ls .devbase
+build.yaml
+e2e.yaml
+docker.yaml
+```
+
+**Note**: For shared configuration, e.g. `docker` (which could be shared between `docker-build` and `docker-push`) that's also allowed and can become a unified target.
```

Comment:
I could see having multiple ways to share configs quickly getting out of had. One way to handle this would be to have a single shared variable file and have a hierarchy where all targets get the shared file, but can supersede these options in their specific config files.

This might result in a big shared variable file, but it keeps things simple until we need the complexity.

Reply:
PTAL at the new wording that expresses what I meant to say here